Workbooks/Azure Logic Apps/B2B Tracking/LAB2bTracking.workbook

```diff
@@ -0,0 +1,1069 @@
sful\\\",\\\"Failed\\\"),\\r\\n    FunctionalAckDirection ,   // 0 Send, 1 Receive \\r\\n    FunctionalAckTime,\\r\\n    FunctionalAckClientTrackingId,\\r\\n    FunctionalAckSubscriptionId,\\r\\n    FunctionalAckResourceGroup,\\r\\n    FunctionalAckActionName,\\r\\n    FunctionalAckLogicAppName,\\r\\n    AckActionName = iff(isnotempty( TransactionSetAckActionName), TransactionSetAckActionName, FunctionalAckActionName),\\r\\n    AckLogicAppName = iff(isnotempty( TransactionSetAckLogicAppName), TransactionSetAckLogicAppName, FunctionalAckLogicAppName),\\r\\n    AckDirection = iff(isnotempty(TransactionSetAckDirection), TransactionSetAckDirection, FunctionalAckDirection),\\r\\n    AckTime = iff(isnotempty(TransactionSetAckTime), TransactionSetAckTime, FunctionalAckTime)\\r\\n    | extend  IsAckValid = AckTime > MessageTime\\r\\n    | extend MessageStatus = iff(TransactionSetStatus == \\\"Successful\\\" and IsFunctionalAckExpected ,\\r\\n     iff(isnotempty(TransactionSetAckStatusCode) and IsAckValid, TransactionSetAckStatus, iff(isnotempty(FunctionalAckStatusCode) and IsAckValid, FunctionalAckStatus, \\\"Pending\\\")),\\r\\n      TransactionSetStatus)\\r\\n    | extend AckStatus = iff(IsFunctionalAckExpected == true , iff(isnotempty( TransactionSetAckStatusCode) and IsAckValid, TransactionSetAckStatusCode,\\r\\n       iff (isnotempty( FunctionalAckStatusCode) and IsAckValid, FunctionalAckStatusCode, \\\"Pending\\\")), \\\"NotRequired\\\"))\\r\\n    | union \\r\\n    (EdiTrackRecords\\r\\n        | where EventTime {TimeRange}\\r\\n        | where RecordType == \\\"X12TransactionSet\\\"\\r\\n        | where isempty(TransactionSetControlNumber)\\r\\n        | where MessageProperties.MessageType !in (\\\"997\\\", \\\"999\\\")\\r\\n        |project \\r\\n            AgreementName,\\r\\n            SenderPartnerName,\\r\\n            ReceiverPartnerName,\\r\\n            MessageClientTrackingId = ClientRequestId,\\r\\n            InterchangeControlNumber,\\r\\n            FunctionalGroupControlNumber,\\r\\n            TransactionSetControlNumber,\\r\\n            MessageTime = EventTime,\\r\\n            MessageSubscriptionId = WorkflowRunOperationInfo.Workflow.SubscriptionId,\\r\\n            MessageResourceGroup =  WorkflowRunOperationInfo.Workflow.ResourceGroup,\\r\\n            MessageActionName =  WorkflowRunOperationInfo.Operation.OperationName,\\r\\n            MssageLogicAppName = WorkflowRunOperationInfo.Workflow.Name,\\r\\n            MessageDirection = Direction,\\r\\n            IsFunctionalAckExpected = MessageProperties.IsFunctionalAcknowledgmentExpected\\r\\n            | extend MessageStatus = \\\"Failed\\\")\\r\\n    | summarize count() by MessageStatus;\\r\\n    let NameColumn = datatable(MessageStatus:string)[\\\"Successful\\\", \\\"Pending\\\", \\\"Failed\\\"];\\r\\nNameColumn\\r\\n| lookup queryResult on MessageStatus \\r\\n| extend count = coalesce(['count_'], 0)\",\"clusterName\":\"{KustoCluster}\",\"databaseName\":\"{Database}\"}",
```

Comment:
Very difficult to review queries like this. Does the repo has no mean to keep the query separated from the json file so it may be formatted more traditionally and easily reviewed in a suitable IDE?